Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModel.kt

```diff
@@ -152,11 +155,10 @@ class PersonOverviewViewModel @AssistedInject constructor(
         null
     }
 
-    fun refreshCertificate(containerId: TestCertificateContainerId) =
-        launch {
-            val error = testCertificateRepository.refresh(containerId).mapNotNull { it.error }.singleOrNull()
-            error?.let { events.postValue(ShowRefreshErrorDialog(error)) }
-        }
+    fun refreshCertificate(containerId: TestCertificateContainerId) = launch(scope = appScope) {
+        val error = testCertificateRepository.refresh(containerId).mapNotNull { it.error }.singleOrNull()
+        error?.let { events.postValue(ShowRefreshErrorDialog(error)) }
+    }
```

Comment:
While the `refresh` call that happens automatically, happens on a worker, that should not be canceled mid process, the refresh call via the button in the UI runs the `viewModelScope` which is canceled when the screen is left.

But:
I don't think it's the case anymore, switching this to `AppScope` doesn't hurt.

The key registration itself happens via `HotData.updateBlocking` which passes the execution to the scope of `HotData` which already is `AppScope`:

```kotlin
    suspend fun updateBlocking(action: suspend T.() -> T): T {
        val update: Update<T> = Update(onModify = action)
        updateActions.tryEmit(update) // Passes the action to be executed on a different scope

        Timber.tag(tag).v("Waiting for update.")
        val ourUpdate = internalFlow.first { it.updatedBy == update } // Only waiting for this would be canceled

        ourUpdate.error?.let { throw it }

        return ourUpdate.value
    }
```